lib/lambda_ethereum_consensus/state_transition/accessors.ex

```diff
@@ -121,4 +121,53 @@ defmodule LambdaEthereumConsensus.StateTransition.Accessors do
     |> Stream.map(fn {_validator, index} -> index end)
     |> Enum.to_list()
   end
+
+  @doc """
+  Return the seed at ``epoch``.
+  """
+  @spec get_seed(BeaconState.t(), SszTypes.epoch(), SszTypes.domain_type()) :: SszTypes.bytes32()
+  def get_seed(state, epoch, domain_type) do
+    mix =
+      get_randao_mix(
+        state,
+        epoch + ChainSpec.get("EPOCHS_PER_HISTORICAL_VECTOR") -
+          ChainSpec.get("MIN_SEED_LOOKAHEAD") - 1
+      )
+
+    :crypto.hash(:sha256, domain_type <> Misc.uint_to_bytes4(epoch) <> mix)
+  end
+
+  @doc """
+  Return the beacon proposer index at the current slot.
+  """
+  @spec get_beacon_proposer_index(BeaconState.t()) :: SszTypes.validator_index()
+  def get_beacon_proposer_index(state) do
+    epoch = get_current_epoch(state)
+
+    seed =
+      :crypto.hash(
+        :sha256,
+        get_seed(state, epoch, Constants.domain_beacon_proposer()) <>
+          Misc.uint_to_bytes4(state.slot)
+      )
+
+    indices = get_active_validator_indices(state, epoch)
+    proposer_index = Misc.compute_proposer_index(state, indices, seed)
+    proposer_index
+  end
+
+  @doc """
+  Return the signature domain (fork version concatenated with domain type) of a message.
+  """
+  @spec get_domain(BeaconState.t(), SszTypes.domain_type(), SszTypes.epoch() | None) :: SszTypes.domain()
+  def get_domain(state, domain_type, epoch \\ None) do
+    epoch = if epoch == None, do: get_current_epoch(state), else: epoch
```

Comment:
`None` in Python is equivalent to `nil` in Elixir

```suggestion
    epoch = if epoch == nil, do: get_current_epoch(state), else: epoch
```